### PR TITLE
Editor: Add Semantic Headings to the right click menu of Text Elements

### LINKED ISSUE #11402: Editor: Add Semantic Headings to the right click menu of Text Elements
## Feature Description

Now that automatic semantic headings are applied to text elements within stories (see https://github.com/googleforcreators/web-stories-wp/issues/5326), users are requesting a way to be able to view what that value is or will be on publish so adjustments can be made beforehand.

This feature will allow a user to select the text within a page and see and change the text element within the sidebar.

Many thanks to @swissspidy for the implementation idea, and to @aaskedall for the UX guidance.

## Acceptance Criteria

- This should be under a new experimental feature flag.
  -  The semantic headings experiment flag is on the branch titled: `feature/11788-experiment-semantic-headings`... Please branch off of this when working on this feature.
  - For an example of how to use it with a feature, see this PR: https://github.com/GoogleForCreators/web-stories-wp/pull/11626/files
- Within the right click menu of a text element on the page, there should be a new item called Heading Level
- This Heading Level item should have a submenu that show the following properties (along with the one currently selected):
	- `Automatic (xyz)` (stored as `auto`)
	- `Heading 1` (stored as `h1`)
	- `Heading 2` (stored as `h2`)
	- `Heading 3` (stored as `h3`)
	- `Paragraph` (stored as `p`)
- When the menu item is changed to another option in the menu, it should update the text element's `tagName property` 

## Developer Notes

Notes from @swissspidy  on implementation:
`xyz` is replaced with whatever `getTextElementTagNames()` returns for the element.

When changing this option, the text element's `tagName` property is updated.

`getTextElementTagNames()` is updated to take this `tagName` property into account.

## Alternatives Considered

In addition to this space, this functionality will also be accomplished within the sidebar of the Editor.

### PR DESCRIPTION
**Pull request by timarney**
## Context

Within the right click menu of a text element on the page, there should be a new item called Heading Level
This Heading Level item should have a submenu that shows the following properties (along with the one currently selected):
- `Automatic (xyz)` (stored as auto)
- `Heading 1` (stored as `h1`)
- `Heading 2` (stored as `h2`)
- `Heading 3` (stored as `h3`)
- `Paragraph` (stored as `p`)

## Summary

Adds right click menu for heading level to text elements.

Screen Shot 2022-08-03 at 5 39 10 AM

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Add a text element to the canvas
2. Right click the element
3. Select Heading Level from the `Right Click Menu`.
4. Select a new heading level from the sub menu and note that it gets selected when click
5. Verify the style panel heading level matches what is selected from the `Right Click Menu`.
6. Set the heading level to `Automatic` - note that the "current" tag name is noted for the "Automatic" option

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11402

### CODE REVIEW

File: packages/story-editor/src/app/rightClickMenu/hooks/useHeadingSelect.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      setIsSubMenuOpen(false);
+    }
+  }, [isMenuOpen]);
+
+  const subMenuItems = useMemo(() => {
+    if (!isMenuOpen || selectedElements.length === 0) {
+      return [];
+    }
+
+    const HEADING_LEVELS = {
```
Comment (miina) [inline packages/story-editor/src/app/rightClickMenu/hooks/useHeadingSelect.js]: Nit: we can move this out of the function for consistency, we generally keep the constants defined like this with capital letters out of the functions.

**Review by miina — APPROVED** [review]
LGTM!

Would be nice to add this to the Storybook as well for the text element, to keep it up to date.

File: packages/story-editor/src/app/rightClickMenu/hooks/useHeadingSelect.js (hunk 1 above)
Comment (timarney) [reply packages/story-editor/src/app/rightClickMenu/hooks/useHeadingSelect.js]: Moved

https://github.com/GoogleForCreators/web-stories-wp/pull/12062/commits/aa981e4b0eda565f36f914645c9b03c7788b09ec#diff-a587530b32e93f39df70710cc55d95daffe62546e81293de4d2e64fa441e0f19R36

**Comment by timarney** [comment]
Story added.

Screen Shot 2022-08-04 at 7 36 22 AM

**Pull request merged**